src/shared/utils/formatTime.ts

```
@@ -0,0 +1,292 @@
+
+export type DatePickerFormat = Dayjs | Date | string | number | null | undefined
+
+export const formatTimePatterns = {
+  dateTime: "DD MMM YYYY h:mm a", // 17 Apr 2022 12:00 am
+  date: "DD MMM YYYY", // 17 Apr 2022
+  time: "h:mm a", // 12:00 am
+  split: {
+    dateTime: "DD/MM/YYYY h:mm a", // 17/04/2022 12:00 am
+    date: "DD/MM/YYYY", // 17/04/2022
+  },
+  paramCase: {
+    dateTime: "DD-MM-YYYY h:mm a", // 17-04-2022 12:00 am
+    date: "DD-MM-YYYY", // 17-04-2022
+  },
+}
+
+const isValidDate = (date: DatePickerFormat) =>
+  date !== null && date !== undefined && dayjs(date).isValid()
+
+// ----------------------------------------------------------------------
+
+/**
+ * Utility functions for formatting times
+ */
+export const FormatTime = {
+  /**
+   * Returns today's date
+   * @returns {string}
+   */
+  today(template?: string): string {
+    return dayjs(new Date()).startOf("day").format(template)
+  },
+
+  /**
+   * Formats a date into human-readable date-time string
+   * @output 17 Apr 2022 12:00 am
+   */
+  dateTime(date: DatePickerFormat, template?: string): string {
+    if (!isValidDate(date)) {
+      return "Invalid date"
+    }
+
+    return dayjs(date).format(template ?? formatTimePatterns.dateTime)
+  },
+
+  /**
+   * Formats a date into a short date string
+   * @output 17 Apr 2022
+   */
+  date(date: DatePickerFormat, template?: string): string {
+    if (!isValidDate(date)) {
+      return "Invalid date"
+    }
+
+    return dayjs(date).format(template ?? formatTimePatterns.date)
+  },
+
+  /**
+   * Formats a date into a time string
+   * @output 12:00 am
+   */
+  time(date: DatePickerFormat, template?: string): string {
+    if (!isValidDate(date)) {
+      return "Invalid date"
+    }
+
+    return dayjs(date).format(template ?? formatTimePatterns.time)
+  },
+
+  /**
+   * Returns a UNIX timestamp (milliseconds) from a date
+   * @output 1713250100
+   */
+  timestamp(date: DatePickerFormat): number | "Invalid date" {
```

It's better to return an exception.